dangerfile.ts

```diff
@@ -37,3 +40,53 @@ if (!hasChangelog) {
     }
   })
 }
+
+/* === Package dependencies ========================================================================================= */
+
+let dependenciesChangedLogged = false
+
+function logDependenciesChanged() {
+  if (!dependenciesChangedLogged) {
+    dependenciesChangedLogged = true
+    warn('Package dependencies changed. Make sure you have OSS approval before merging!')
+  }
+}
+
+danger.git.created_files
+  .filter(filepath => filepath.match(/\bpackage\.json$/))
+  .forEach(filepath => {
+    warn(`New package added: ${filepath}. Make sure you have OSS approval before merging!`)
+  })
+
+danger.git.modified_files
+  .filter(filepath => filepath.match(/\bpackage\.json$/))
+  .forEach(async filepath => {
```

Comment:
```js
danger.git.modified_files
  .filter(filepath => filepath.match(/\bpackage\.json$/))
  .reduce(async (acc, filepath) => {
    const changedDependencies = await getChangedDependencies(filepath)

    if (Object.keys(changedDependencies).length > 0) {
      acc[filename] = changedDependencies
    }

    return acc
  }, {})
  .reduce((hasWarning, changedDependencies, filename) => {
    // create markdown
    markdown(changesTabled)

    if (hasWarning) return true
    warn('Package dependencies changed. Make sure you have approval before merging!')
  }, false)
```

I expect something like this, less functions and more clear steps
/cc @kuzhelov